src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -51,33 +56,60 @@ private void CompilationStartAction(CompilationStartAnalysisContext context)
                 // actually necessary.
                 var viewModel = SerializableNamingStylePreferencesInfo.FromXElement(XElement.Parse(currentValue));
                 var preferencesInfo = viewModel.GetPreferencesInfo();
+
+                var idToCachedResult = new Dictionary<Guid, ConcurrentDictionary<string, string>>();
```

Comment:
@pharring would say that `ConcurrentDictionary` is a pretty bulky object.  It may be the right thing to use here given the number of identifiers, but we should measure a couple of variations.